add new pollingInterval & cooldownPeriod tests

gauron99 opened the pull request:
- New `internals/` tests for PI & CdP
- I also created new helper.go function thats exactly same as `kubectl apply` but is `kubectl create` instead because of the job for updating metrics - when using apply, the test runs so fast that the job doesnt have time to get created, finished and deleted before the next apply and returns an error(job immutable). Because of this i used generated Job names as well.
- Sometimes there is a timing issue with `AssertReplicaCountNotChangeDuringTimePeriod` that the value changes before the 60 seconds (like 59s) therefore i added 2 seconds to the cooldowns to ensure it passes.
### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4266 
Fixes #4267

zroubalik commented:
/run-e2e polling_cooldown*

zroubalik commented:
/run-e2e polling_cooldown*

JorTurFer commented on tests/internals/polling_cooldown_so/polling_cooldown_so_test.go:
```diff
@@ -0,0 +1,309 @@
+`
+)
+
+// use KubectlCreateWithTemplate() function because applying yaml template
+// for update-metrics relies on the previous job to be deleted
+// (job is immutable), but in this case the test runs too fast at some
+// points that job is not finished & deleted in time
```
Isn't`KubectlApplyWithTemplate` with `ttlSecondsAfterFinished: 0` enough for removing the job?

gauron99 replied:
its not, the job has a few seconds to run and it didnt finish in time.

JorTurFer commented on tests/internals/polling_cooldown_so/polling_cooldown_so_test.go:
```diff
@@ -0,0 +1,309 @@
+	MinReplicas                 string
+	MaxReplicas                 string
+	MetricValue                 int
+	PollingInterval             string
+	CooldownPeriod              string
```
Can we use `int` instead of `string`?

JorTurFer reviewed (commented):
LGTM, some comments inline

gauron99 commented:
i changed string->int as suggested and also changed reserve time to 5 seconds because one of the times of my testing it still failed, so to avoid random failed runs, 5 sec should be enough

JorTurFer reviewed (approved):
LGTM!

JorTurFer commented:
/run-e2e polling_cooldown*

JorTurFer commented:
/run-e2e polling_cooldown*

JorTurFer commented:
/run-e2e polling_cooldown*
